### PR TITLE
search: move streaming HTTP events to internal/search/streaming/http

### PR DESCRIPTION
**Pull request by keegancsmith**
Will likely end up in a better place at some point, but this seems like
a reasonable pkg name at first. Will likely also add a client/reader to
this pkg which can be used by integration testing/etc.

### CODE REVIEW

**Review by stefanhengl — APPROVED** [review]
Almost feels like http and streaming should be flipped. I am not sure though.

streaming/http/ -> http/streaming/

File: internal/search/streaming/http/doc.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// package http contains Sourcegraph's streaming HTTP protocol, which is based
+// on Server Sent Events (SSE).
+package http
```
[the remark this answers is not in the document]
Comment (keegancsmith) [reply internal/search/streaming/http/doc.go:3]: `doc.go` is the standard name for a go file which just has a package docstring.

**Comment by keegancsmith** [comment]
> Almost feels like http and streaming should be flipped. I am not sure though.
> 
> streaming/http/ -> http/streaming/

Yeah our pkg layout is a bit jumbled at the moment. I think once we are happy with exhaustive + the work Camden is doing on extracting out of graphql we should take a moment to do some high level movements of code so the pkg hierarchy feels nice.

File: internal/search/streaming/http/doc.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefanhengl) [reply internal/search/streaming/http/doc.go:3]: 🤦 That makes more sense now.

**Pull request merged**